src/pages/api/conversation.ts

```diff
@@ -0,0 +1,191 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import * as deepl from "deepl-node";
+import { VALID_LANGUAGE_CODES, TargetLanguageCode } from "@/utils/languages";
+import {
+  generateConversationResponse,
+  ConversationMessage,
+  isOpenAIConfigured,
+} from "@/lib/openai";
+
+const translator = new deepl.Translator(process.env.DEEPL_API_KEY || "");
+
+type ConversationRequestBody = {
+  userMessage: string;
+  targetLanguage: string;
+  conversationHistory?: ConversationMessage[];
+};
+
+type ConversationSuccessResponse = {
+  userMessageTranslation: string;
+  assistantResponse: string;
+  assistantResponseTranslation: string;
+};
+
+type ErrorResponse = {
+  message: string;
+  fallback?: boolean;
+};
+
+/**
+ * Conversation API endpoint
+ * 
+ * Implements a 2-step flow:
+ * 1. Generate AI response in user's original language
+ * 2. Batch translate both user message and AI response to target language
+ * 
+ * Falls back to translation-only mode if OpenAI is unavailable
+ */
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ConversationSuccessResponse | ErrorResponse>
+) {
+  if (req.method !== "POST") {
+    res.status(405).json({ message: "Method not allowed" });
+    return;
+  }
+
+  const data = req.body as ConversationRequestBody;
+
+  // Validate required fields
+  if (!data.userMessage) {
+    res.status(400).json({ message: "userMessage is required" });
+    return;
+  }
+
+  if (!data.targetLanguage) {
+    res.status(400).json({ message: "targetLanguage is required" });
+    return;
+  }
+
+  // Validate target language
+  if (!VALID_LANGUAGE_CODES.has(data.targetLanguage)) {
+    res.status(400).json({
+      message: `Invalid target language: ${data.targetLanguage}. Please provide a valid language code.`,
+    });
+    return;
+  }
```

Comment:
This feels pretty verbose. Can we make it a little more concise? Some stuff must be shareable here, no?